### PR TITLE
Re-add index_per_partition but as deprecated

### PR DESCRIPTION
**Pull request by maysamyabandeh**
index_per_partition should have deprecated deprecated instead of being removed. It is causing backward compatibility issues.

### CODE REVIEW

File: include/rocksdb/table.h [hunk 1]
```diff
@@ -155,6 +155,9 @@ struct BlockBasedTableOptions {
   // separately
   uint64_t metadata_block_size = 4096;
 
+  // Deprecated
+  uint64_t index_per_partition = 1024;
```
Comment (siying) [inline include/rocksdb/table.h]: I don't think we need this. We can remove this.

File: options/options_helper.h [hunk 2]
```diff
@@ -600,6 +600,9 @@ static std::unordered_map<std::string, OptionTypeInfo>
         {"index_block_restart_interval",
          {offsetof(struct BlockBasedTableOptions, index_block_restart_interval),
           OptionType::kInt, OptionVerificationType::kNormal, false, 0}},
+        {"index_per_partition",
+         {offsetof(struct BlockBasedTableOptions, index_per_partition),
```
Comment (siying) [inline options/options_helper.h]: once we remove index_per_partition, you can set 0 here.

**Comment by maysamyabandeh** [comment]
I can create a db without the patch and open it with the patch. Would that work?

**Comment by siying** [comment]
If you can call LoadLatestOptions() successfully against a DB generated by the previous release, it will be sufficient.

**Comment by siying** [comment]
CC @gunnarku

**Comment by maysamyabandeh** [comment]
I confirm it works with this test and adding the option manually to the option file:
```
293   DBOptions db_opt;
294   std::vector<ColumnFamilyDescriptor> cf_descs;
295   ASSERT_OK(LoadOptionsFromFile(kOptionsFileName, Env::Default(), &db_opt,
296                                 &cf_descs));
297   Options opt(db_opt, cf_descs[0].options);
```

Before the patch:
```
LoadOptionsFromFile(kOptionsFileName, Env::Default(), &db_opt, &cf_descs)
Invalid argument: Can't parse BlockBasedTableOptions:: index_per_partition Unrecognized option
[  FAILED  ] DBBenchTest.OptionsFileFromFile (0 ms)
```

After the patch:
```
[       OK ] DBBenchTest.OptionsFileFromFile (1 ms)
```

**Comment by siying** [comment]
Awesome!

**Pull request closed without merging**